### ISSUE TITLE
Add a flag/property to disable logging compiler errors during analysis

### ISSUE THREAD

**Issue by eygraber** [detekt/detekt#9293]
I try to keep a very clean log, but with Detekt's type resolution I get a log of:

> There were X compiler errors found during analysis

It would be nice if there was a way to stop the logging from happening.

**Comment by 3flex** [comment]
This warning indicates that your detekt config is likely incorrect for some reason, so analysis quality is compromised.

It should not appear for any tasks that are automatically configured by the Gradle plugin. If it does, we need to investigate and fix the root cause.

If it occurs for tasks that were manually configured then the warning suggests that the config needs to be corrected anyway... Or, analysis should run in the simple mode without semantic analysis.

**Comment by eygraber** [comment]
I ran with `debug = true` and the only thing I see are a bunch of lines that say:

```
The rule '...' requires type resolution but it was run without it
```

If I had to guess I'd say that's when I run Detekt on KMP targets that don't support type resolution (iOS and common in this case). Type resolution is running correctly for the Android target.

**Comment by 3flex** [comment]
> Type resolution is running correctly for the Android target.

That's the task or tasks that are showing the originally reported issue? Not the detekt tasks generated for the iOS or common targets?

**Comment by eygraber** [comment]
It doesn't say what tasks are causing the issue IIRC

**Comment by 3flex** [comment]
Please refer to my original comment.

If it's a task that's configured by the plugin then we should identify and fix the root cause if possible. If it's manually registered then please check the config to make sure it's correct. 

If it's because there's a compiler plugin that generates code, that's a known issue (will link to issue when not on my phone). Likewise for code generated by the Android plugin like view bindings or buildconfig. We can track those fixes under specific issues.

Fixing the root cause would require a reproducer if it's anything else. Otherwise the warning is valid and on balance it's best to keep it.
